src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -138,7 +138,12 @@ def newQueueElement(self, **args):
         ele = WorkQueueElement(**args)
         for data, sites in viewitems(ele['Inputs']):
             if not sites:
-                raise WorkQueueWMSpecError(self.wmspec, 'Input data has no locations "%s"' % data)
+                # make call to rucio to get list of replicas for this input data
+                locations = self.getDatasetLocations([data])
```

Comment:
This method is meant to be used only for pileup datasets. Looking at the implementation, it would work for a block as well (rucio dataset), but then it would be using a wrong rucio account, as can be seen in this line:
https://github.com/dmwm/WMCore/blob/baa873418766654f571d3d11be16fcc2bb3cc6f3/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py#L265

We have basically 2 alternatives:
1) remove this call; or
2) change the `getDatasetLocations` implementation such that we can resolve the location either with `self.rucioAcct` or `self.rucioAcctPU`, according to the data type (pileup for the latter, else the former account).